### PR TITLE
deployer: add mock builder

### PR DESCRIPTION
**Pull request by iulianbarbu**
## Description of change

1) Added a `MockedBuilder` struct that receives a source code archive and sends back a default docker image archive, provided to the deployer through a binary argument.
2) Added a README guiding through how to build such a docker image archive.
3) Prepared the `openapi` endpoints by adding the the structs used to serve the new deployer APIs docs.
4) Changed the endpoint for deploying a project to `/project/:project_name`, similarly to how it was in the past.

## How Has This Been Tested (if applicable)?

Started the deployer instructing it to read the image archive and fetched it as if the deployer handled a call to deploy a project.

### CODE REVIEW

File: cargo-shuttle/src/client.rs [hunk 1]
```diff
@@ -32,7 +32,7 @@ impl Client {
     }
 
     pub async fn deploy(&self, data: Vec<u8>, project: &ProjectName) -> Result<String> {
-        let path = format!("/deploy/{}", project.as_str(),);
+        let path = format!("/project/{}", project.as_str(),);
```
Comment (chesedo) [inline cargo-shuttle/src/client.rs:35]: I was fine with the new name. Maybe should have an API route discussing with the next cycle plan

File: deployer/src/api/mod.rs [hunk 2]
```diff
@@ -37,5 +44,12 @@ pub async fn deploy_project(
     }
     debug!("Received a total of {} bytes", data.len());
 
+    debug!("Seding project source code to the builder.");
```
Comment (chesedo) [inline deployer/src/api/mod.rs:47]:
```suggestion
    debug!("Sending project source code to the builder.");
```

File: deployer/src/builder/README [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+4) Run:
+```bash
+docker save shuttle-service > shuttle-service.tar
+```
+
+5) Optionally, you can remove the shuttle-service docker image:
+```bash
+docker rmi shuttle-service
+```
```
Comment (chesedo) [inline deployer/src/builder/README:23]: You can use `--output` to build to a .tar directly

https://docs.docker.com/engine/reference/commandline/build/#output

**Review by chesedo — APPROVED** [review]
Looks good, thanks Iu!!

File: deployer/src/builder/README (hunk 3 above)
Comment (iulianbarbu) [reply deployer/src/builder/README]: Done.

File: deployer/src/api/mod.rs (hunk 2 above)
Comment (iulianbarbu) [reply deployer/src/api/mod.rs]: Done.

File: deployer/Cargo.toml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "shuttle-deployer"
-version = "0.16.0"
+version = "1.0.0"
```
Comment (oddgrd) [inline deployer/Cargo.toml:3]:
```suggestion
version = "0.1.0"
```

**Review by oddgrd — APPROVED** [review]
LGTM!

File: deployer/Cargo.toml (hunk 4 above)
Comment (oddgrd) [reply deployer/Cargo.toml:3]: Or "0.16.0" for simplicity :shrug:

**Pull request merged**